lib/src/command/outdated.dart

```diff
@@ -185,14 +224,42 @@ class OutdatedCommand extends PubCommand {
     }
   }
 
+  /// Get the latest version of [ref] given its [PackageRef].
+  ///
+  /// Returns `null`, if unable to find the package.
+  Future<PackageId> _getLatest(PackageRef ref) async {
```

Comment:
As far as I can see you always call this with `id.toRef()`.
Maybe it could be
```
Future<PackageId> _getLatestFromSameSource(PackageId id) {
  if (id == null) return null;
  ref = id.toRef();
  ...
}
```